### PR TITLE
feat: foundational ffmpeg transcoding

### PR DESCRIPTION
**Pull request by hbomb79**
Manual creation and tear-down (cancellation and cleanup) of FFmpeg transcodings via REST client, along with automatic cancellation of running transcodes when Thea is shutdown.

### CODE REVIEW

File: internal/media/store.go [hunk 1]
```diff
@@ -90,12 +91,12 @@ const (
...
 	if err := db.QueryRowx(`
-		INSERT INTO movie(id, tmdb_id, title, adult, source_path, created_at, updated_at)
-		VALUES($1, $2, $3, $4, $5, current_timestamp, current_timestamp)
+		INSERT INTO media(id, type, tmdb_id, title, adult, source_path, created_at, updated_at)
+		VALUES($1, $2, $3, $4, $5, $6, current_timestamp, current_timestamp)
 		ON CONFLICT(tmdb_id) DO UPDATE
-			SET (updated_at, title, adult, source_path) = (current_timestamp, EXCLUDED.title, EXCLUDED.adult, EXCLUDED.source_path)
-		RETURNING *
-	`, movie.ID, movie.TmdbId, movie.Title, movie.Adult, movie.SourcePath).StructScan(&updatedMovie); err != nil {
+			SET (updated_at, title, adult, source_path, type) = (current_timestamp, EXCLUDED.title, EXCLUDED.adult, EXCLUDED.source_path, 'movie')
```
Comment (hbomb79) [inline internal/media/store.go]: Hm, just realised this is a problem. Using only the tmdb_id to detect conflicts sounds like a good idea, but today while messing with their API I discovered their IDs are only unique in their category (i.e. movies and series with the same ID do exist). We should probably make this constraint more robust by only considering a row to be conflicting if it has the same ID **and** type (movie|episode).

File: internal/database/migrations/0001_initial.sql
Comment (hbomb79) [inline internal/database/migrations/0001_initial.sql:1]: Modifying this initial state is naughty for sure, but also... this isn't deployed anywhere so idgaf

File: internal/ffmpeg/exec.go [hunk 2]
```diff
@@ -112,7 +124,7 @@ func (cmd *TranscodeCmd) OutputPath() string {
 }
 
 func (cmd *TranscodeCmd) String() string {
-	var pid int = -1
+	pid := -1
```
Comment (AlDrag) [inline internal/ffmpeg/exec.go:127]: Prefer `:=` in most cases?

I do like it.

File: internal/http/tmdb/search.go [hunk 3]
```diff
@@ -299,24 +299,24 @@ type (
 		StatusMessage string `json:"status_message"`
 	}
 	FailedRequestError struct {
-		httpCode int
-		tmdbCode int
-		message  string
+		HttpCode int
+		TmdbCode int
+		Message  string
```
Comment (AlDrag) [inline internal/http/tmdb/search.go]: Why are these public now?

**Review by AlDrag — APPROVED** [review]
Just gave it a quick scan to see how the code looks.

Obviously understanding the entire context would take an entire day, but it's looking super clean! ❤️

File: internal/media/store.go (hunk 1 above)
Comment (AlDrag) [reply internal/media/store.go]: Should be simple to just modify the constraint for this PR right?

File: internal/http/tmdb/search.go (hunk 3 above)
Comment (hbomb79) [reply internal/http/tmdb/search.go]: Hm, that's a good question. I've made them private (and in the other errors too). Hopefully nothing subtle breaks, but from what I can tell the `String()` implementation is all that uses the error fields (which is internal to the package anyway).

File: internal/ffmpeg/exec.go (hunk 2 above)
Comment (hbomb79) [reply internal/ffmpeg/exec.go:127]: Haha yeah, cleaner than an if-else block to initialise it IMO :)

File: internal/media/store.go (hunk 1 above)
Comment (hbomb79) [reply internal/media/store.go]: Yeah fair enough, I'll update the initial schema UK and the relevant query now

Comment (hbomb79) [reply internal/media/store.go]: Okay, updated this to look for conflicts over both `tmdb_id` and `type`. Also replaced the simple column `UNIQUE` contraint with a composite unique constraint.

Comment (AlDrag) [reply internal/media/store.go]: New commit looks good!

**Comment by hbomb79** [comment]
Thanks @AlDrag , yeah there's a LOT of context to un-pack here. It's no small project for sure, but I'm glad you found the code clean/readable given the complexity of what's being handled. There's still a lot of rough edges/corner cases to tackle, but as an MVP I'm just handling what is easy for now haha

**Pull request merged**